src/ray/core_worker/lib/golang/go_worker.cpp

```diff
@@ -0,0 +1,328 @@
+#include "go_worker.h"
+
+#include <stdint.h>
+
+#include <iostream>
+
+#include "ray/common/id.h"
+#include "ray/core_worker/core_worker.h"
+#include "ray/gcs/gcs_client/global_state_accessor.h"
+
+using namespace std;
+
+template <typename ID>
+inline ID ByteArrayToId(char *bytes) {
+  std::string id_str(ID::Size(), 0);
+  memcpy(&id_str.front(), bytes, ID::Size());
+  return ID::FromBinary(id_str);
+}
+
+DataBuffer *AllocateDataBuffer(void *ptr, int size) {
+  auto db = new DataBuffer();
+  db->p = ptr;
+  db->size = size;
+  return db;
+}
+
+RAY_EXPORT DataValue *go_worker_AllocateDataValue(void *data_ptr, size_t data_size,
+                                                  void *meta_ptr, size_t meta_size) {
+  auto dv = new DataValue();
```

Comment:
Are you sure that C++ will free DataValue created this way when passed to C++ as a ptr? I believe it will not, right? How should it be managed, then?